src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -678,6 +700,10 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
                     self.add(Qualif::STATIC_REF);
                 }
 
+                if self.mode == Mode::ConstFn && self.tcx.sess.features_untracked().const_let {
+                    return;
```

Comment:
This is bad because it breaks promotion in `const fn`. Also, you want mutable borrows to locals outside of `const fn` too, right? I think we should orthogonally allow `&mut foo` in regular `static`, so only `const` needs a special-case at the outermost level (and promotion also needs to keep disallowing `&mut NotZST`).